src/pages/index.js

```diff
@@ -1,130 +1,223 @@
+import { UserInfo } from "../components/UserInfo";
+import { ConfirmationPopup } from "../components/ConfirmationPopup";
+import * as Constants from "../utils/Constants";
 
 (function displayCards() {
-  // Elements
-  const imageClickHandler = ({ name, link }) => {
-    imagePopup.open({ name, link });
-  };
-  const cardTemplate =
-    document.querySelector("#card-template").content.firstElementChild;
-  const galleryList = document.querySelector("#gallery__list");
-
-  const items = [
-    {
-      name: "Yosemite Valley",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/yosemite.jpg",
+  const api = new Api({
+    baseUrl: "https://around-api.en.tripleten-services.com/v1",
+    headers: {
+      authorization: "9b29ae94-bb6f-470b-bccd-5f1bdf13a16a",
+      "Content-Type": "application/json",
     },
-
+  });
+  let initialCardsInfo = NaN;
+  const cardsSection = new Section(
     {
-      name: "Lake Louise",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lake-louise.jpg",
+      initialCardsInfo,
+      renderer: (cardData) => {
+        addCard(
+          cardData.name,
+          cardData.link,
+          Constants.cardTemplate,
+          cardData._id
+        );
+      },
     },
+    Constants.galleryList
+  );
 
-    {
-      name: "Bald Mountains",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/bald-mountains.jpg",
-    },
+  api
+    .getInitialcards()
+    .then((items) => {
+      initialCardsInfo = items.reverse();
+      cardsSection.items = items;
+      cardsSection.renderItems();
+    })
+    .catch((error) => {
+      console.error("Caught an error: ", error.message);
+    });
 
-    {
-      name: "Latemar",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/latemar.jpg",
-    },
+  const editAvatarPopup = new PopupWithForm("#avatar-modal", chageAvatar);
+
+  function chageAvatar(inputValue) {
+    api
+      .changeAvatar(inputValue.UrlInput)
+      .then(() => {
+        user.setAvatar(inputValue.UrlInput);
+      })
+      .then(() => {
+        editAvatarPopup.close();
+      })
+      .catch((error) => {
+        console.error("Caught an error: ", error.message);
+      })
+      .finally(() => {
+        editAvatarPopup.submitBtn.textContent = "Saving...";
+      });
+  }
 
-    {
-      name: "Vanoise National Park",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/vanoise.jpg",
-    },
+  editAvatarPopup.setEventListeners();
 
-    {
-      name: "Lago di Braies",
-      link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lago.jpg",
-    },
-  ];
+  Constants.avatarPhoto.addEventListener("click", () => {
+    editAvatarPopup.open();
+    editAvatarPopup.submitBtn.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.